src/components/segmentBar.js

```diff
@@ -0,0 +1,50 @@
+import React from "react"
+
+export class SegmentProp {
+  label = ""
+  color = "#ffffff"
+  backgroundColor = "#000000"
+  volume = 0
+
+  constructor(label, color, backgroundColor, volume) {
+    this.label = label
+    this.color = color
+    this.backgroundColor = backgroundColor
+    this.volume = volume
+  }
+}
+
+const SegmentBar = ({ segmentProps, className }) => {
+  const totalVolume = segmentProps
+    .map(p => p.volume)
+    .reduce((prev, curr) => prev + curr, 0)
+
+  return (
+    <div
+      class={className}
+      css={{
+        display: "flex",
+        border: "1px solid black",
+        minHeight: 44,
+        fontFamily: "var(--ff-title)",
+        margin: "2rem",
+      }}
+    >
+      {segmentProps.map(segmentProp => (
+        <div
+          css={{
+            width: (segmentProp.volume / totalVolume) * 100 + "%",
```

Comment:
Regarding the use of `width`, I suggest you to separate it as `style={{ width: ... }}`.

The reason is that when using `css={{}}` it compiles to `class="..."`, while `style={{ }}` compiles to, surprisingly, `style="..."`. Try switching between these two and inspect it for yourself.

That being said, width attribute is kinda dynamic here depending on data, so using "style" is better I think.

Reply:
Thanks for the review! I have reflected your review in the new commits. I am quite a newbie in React right now. Definitely need to do more homework. 😅